inc/computeroperatingsystem.class.php

```diff
@@ -0,0 +1,199 @@
+    * @param $item         CommonGLPI object
+    * @param $tabnum       (default 1)
+    * @param $withtemplate (default 0)
+   **/
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum=1, $withtemplate=0) {
+
+      self::showForComputer($item);
+      return true;
+   }
+
+
+   /**
+    * Print the computer's operating system form
+    *
+    * @param $comp Computer object
+    *
+    * @return Nothing (call to classes members)
+   **/
+   static function showForComputer(Computer $comp) {
+      global $DB;
+
+      if (!Session::haveRight("computer", UPDATE)) {
+        return false;
+      }
+
+      $ID = $comp->fields['id'];
+
+      $colspan = 4;
+
+      echo "<div class='center'>";
+
+      $comp->initForm($ID);
+      $comp->showFormHeader();
+
+      if ($comp->isNewID($ID)) {
+         echo "<input type='hidden' name='computers_id' value='".$ID."'>";
+      }
+
+      echo "<table class='tab_cadre_fixe'>";
+
+      echo "<tr class='headerRow'><th colspan='".$colspan."'>";
+      echo __('Operating system');
+      echo "</th></tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Name')."</td>";
+      echo "<td>";
+      OperatingSystem::dropdown(array('value' => $comp->fields["operatingsystems_id"]));
+      echo "</td>";
+
+      echo "<td>".__('Version')."</td>";
+      echo "<td >";
+      OperatingSystemVersion::dropdown(array('value' => $comp->fields["operatingsystemversions_id"]));
+      echo "</td>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Architecture')."</td>";
+      echo "<td >";
+      OperatingSystemArchitecture::dropdown(array('value'
+                                                 => $comp->fields["operatingsystemarchitectures_id"]));
+      echo "</td>";
+
+      echo "<td>".__('Service pack')."</td>";
+      echo "<td >";
+      OperatingSystemServicePack::dropdown(array('value'
+                                                 => $comp->fields["operatingsystemservicepacks_id"]));
+      echo "</td></tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Kernel version')."</td>";
+      echo "<td >";
+      Html::autocompletionTextField($comp, 'os_kernel_version');
+      echo "</td>";
+
+      echo "<td>".__('Product ID')."</td>";
+      echo "<td >";
+      Html::autocompletionTextField($comp, 'os_licenseid');
+      echo "</td></tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Serial number')."</td>";
+      echo "<td >";
+      Html::autocompletionTextField($comp, 'os_license_number');
+      echo "</td></tr>";
+
+      if (!empty($ID)
+          && Plugin::haveImport()
+          && $comp->fields["is_dynamic"]) {
+         $inventory_show = true;
+      }
+
+      if (Plugin::haveImport() && $inventory_show) {
+         $cos = new ComputerOperatingSystem();
+         $cos->getFromDB($comp->getID());
+         echo '<tr><td colspan="4">';
+         Plugin::doHook("autoinventory_information", $cos);
```

Comment:
Using the same hook for 2 different parts doesn't seems ok.
(it could work as you have created a new class, so the hook cvan check its type, but per first comment, the new class is not needed)